assets/script.js

```diff
@@ -104,94 +139,134 @@ function getArtistInfo(userInput) {
-          'alt="' +
-          response.topalbums.album[i].name +
-          '/></a>' +
-          "</li>"
-        );
-      }
-    });
+          for (i = 0, a = 0; a < 4; i++) {
+            var albumImage = response.topalbums.album[i].image[2]["#text"];
+            var albumName = response.topalbums.album[i].name;
+            // Checking if album has both img and alt text
+            if (!albumImage || !albumName) {
+              continue;
+            } else {
+              // Apending the album
+              $("#albums>ul").append(
+                '<li><a id="albums" class="waves-effect waves-light modal-trigger" href="#album-modal"><img src="' +
+                  albumImage +
+                  'alt="' +
+                  albumName +
+                  "/></a>" +
+                  "</li>"
+              );
+              // Incrimenting album count
+              a++;
+            }
+          }
+        });
 
-    // Getting top tracks
-    var topTrackURL =
-      "http://ws.audioscrobbler.com/2.0/?method=artist.gettoptracks&artist=" +
-      userInput +
-      "&api_key=1cdcc6e0cda44cee6b6571363c390279&format=json";
+        // Getting top tracks
+        var topTrackURL =
+          "http://ws.audioscrobbler.com/2.0/?method=artist.gettoptracks&artist=" +
+          userInput +
+          "&api_key=1cdcc6e0cda44cee6b6571363c390279&format=json";
 
-    $.ajax({
-      url: topTrackURL,
-      method: "GET",
-    }).then(function (response) {
-      $("#top-tracks>ol").html("");
-      for (i = 0; i < 5; i++) {
-        $("#top-tracks>ol").append(
-          '<li><a id="tracks" class="waves-effect waves-light collection-item modal-trigger" href="#track-modal">' +
-          "<span>" +
-          response.toptracks.track[i].name +
-          "</span></a></li>"
-        );
+        $.ajax({
+          url: topTrackURL,
+          method: "GET",
+        }).then(function (response) {
+          $("#top-tracks>ol").html("");
+          for (i = 0; i < 5; i++) {
+            $("#top-tracks>ol").append(
+              '<li><a id="tracks" class="waves-effect waves-light collection-item modal-trigger" href="#track-modal">' +
+                "<span>" +
+                response.toptracks.track[i].name +
+                "</span></a></li>"
+            );
+          }
+        });
+        $("#artist-results-page").show(400);
       }
-    });
+    }
   });
 }
 
 //These apply on page load
 $(function () {
   $("#search-icon").on("click", function () {
     var userInput = $("#search-query").val();
-    console.log("user input is " + userInput);
-    $('.result-page').hide(400);
+    $(".result-page").hide(400);
     callAPI(userInput);
     $("#default-search-input>label").remove();
     $("#album-input>label").toggle();
     $("#artist-input>label").toggle();
   });
 
-  //search on <enter key> pressed
+  // Event listeners
+  // Modal function
+  document.addEventListener("DOMContentLoaded", function () {
```

Comment:
this modal function needs to be at the top or the modals don't work for some reason

Reply:
Outdated, function was scoped within the doc.ready function, now sits at the bottom of the doc. Needs to be with the rest of the event listeners, placed after so it applies last after everything else has been loaded in

Reply:
if thats the case put the function after all of the event listeners right at the bottom

Reply:
That wouldn't follow the basic rules of JS: declare global variables > functions > event listeners.